### PR TITLE
Upgrade JMH gradle plugin to 0.5.0-rc-1

### PR DESCRIPTION
**Pull request by paplorinc**
Fixes https://github.com/gradle/gradle/issues/9618

### CODE REVIEW

File: subprojects/base-services/base-services.gradle.kts [hunk 1]
```diff
@@ -26,19 +26,16 @@ dependencies {
...
-    
+
     testFixturesImplementation(library("guava"))
     testImplementation(testFixtures(project(":core")))
     testRuntimeOnly(library("xerces"))
-    
+
     integTestRuntimeOnly(project(":runtimeApiInfo"))
+
+    jmh("org.bouncycastle:bcprov-jdk15on:1.61")
```
Comment (paplorinc) [inline subprojects/base-services/base-services.gradle.kts]: What's the reason for `jmhImplementation` not working here?
And neither is `library("bouncycastle_provider")`

**Comment by lptr** [comment]
@melix says that he'll attempt to address the JMH relocatability problem I found yesterday. Let's wait to see if he can do another release so we can upgrade directly to that.

File: subprojects/base-services/base-services.gradle.kts (hunk 1 above)
Comment (melix) [reply subprojects/base-services/base-services.gradle.kts]: Can you try with rc2 and see if it fixes the pb?

Comment (paplorinc) [reply subprojects/base-services/base-services.gradle.kts]: Pushed, it's working the same way with rc1 and rc2 -> only with `jmh`, not with `jmhImplementation` (getting `java.lang.NoClassDefFoundError: org/bouncycastle/jcajce/provider/digest/MD5$Digest`)

**Comment by paplorinc** [comment]
Merged, thanks.

**Pull request merged**